source/httpfs/driver.go

```diff
@@ -10,10 +10,21 @@ import (
 	"github.com/golang-migrate/migrate/v4/source"
 )
 
-// driver is a migration source driver for reading migrations from
+// ErrDuplicateMigration is an error type for reporting duplicate migration
+// files.
+type ErrDuplicateMigration struct {
+	Filename string
+}
+
+// Error implements error interface.
+func (e ErrDuplicateMigration) Error() string {
+	return fmt.Sprintf("duplicate migration file: %s", e.Filename)
+}
+
+// Driver is a migration source driver for reading migrations from
 // http.FileSystem instances. It implements source.Driver interface and can be
 // used as a migration source for the main migrate library.
-type driver struct {
+type Driver struct {
```

Comment:
Rename to `HTTPFS` since it doesn't fully implement the `source.Driver` interface.
We should remove the `Open()` method

Reply:
Hmm symbol `httpfs.HTTPFS` looks kind of weird. Maybe we could we call it `httpfs.Migrations` or `httpfs.Source`? Especially if we are not making it a fully fledged source driver. It actually behaves more like the helper struct `source.Migrations`.

Reply:
Made an attempt with `httpfs.Migrator`. Closing this conversation for the time being. We could re-open it in the next review round if needed.